### PR TITLE
Updated mysql role

### PR DESCRIPTION
**Pull request by jbzdak**
Copy of `mysql` role from plebia, with updates

### CODE REVIEW

File: templates/backup-pre-mysql.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+# Exits after first failure
+set -e
+
+TMPDIR="{{ MYSQL_SERVER_BACKUP_DIR }}"
+timestamp=$(date +%Y%m%dT%H:%M:%S)
+
+rm -rf "${TMPDIR}"
+mkdir -p "${TMPDIR}"
+mysql -s -e 'show databases' | tail -n +1 | grep -v performance_schema | grep -v information_schema | while read DBNAME; do mysqldump -uroot --complete-insert --events --single-transaction "${DBNAME}" > "${TMPDIR}/${DBNAME}".sql; done
```
Comment (smarnach) [inline templates/backup-pre-mysql.sh]: What exactly is the intention of `tail -n +1` in this pipe?  Isn't that the same as just `cat`, or omitting the process?

Comment (jbzdak) [inline templates/backup-pre-mysql.sh]: So `mysql -s e show databases` output something like this: 
mysql -s -e 'show databases' -u root -p

```
Database
information_schema
amarokdb
dalite_ng
mysql
oscar_travis
performance_schema
```

tail -n +1 outputs everything but the first line 
grep -v removes two magical schemas that can't be dumped.

Comment (smarnach) [inline templates/backup-pre-mysql.sh]: No, `tail -n +1` print everything starting from the first line, it doesn't skip anything:

```
$ seq 3
1
2
3
$ seq 3 | tail -n +1
1
2
3
```

Comment (jbzdak) [inline templates/backup-pre-mysql.sh]: You are right, this is a noop here. Suprisingly: `mysql -s -e 'show databases' -u root -p` shows database header while `mysql -s -e 'show databases' -u root -p | tail -n +1` does not, which is probably due to the fact that mysql is being smart, and tries to detect interactive prompt. 

I'll use `--skip-column-names` switch instead, which will filter `'Databases'`column (just as a means of future-proffing).

**Comment by antoviaque** [comment]
:+1:

**Pull request merged**